xrmocap/utils/data_convert_utils.py

```diff
@@ -0,0 +1,248 @@
+# yapf: disable
+import logging
+import numpy as np
+from enum import Enum
+from mmhuman3d.data.data_structures.human_data import HumanData
+from typing import Optional, Union
+
+from xrmocap.data_structure.body_model import SMPLData, SMPLXData
+
+# yapf: enable
+
+
+class SMPLDataTypeEnum(str, Enum):
+    SMPLDATA = 'smpl data'
+    HUMANDATA = 'human data'
+    AMASS = 'AMASS'
+    UNKNOWN = 'unknown'
+
+
+def validate_shape(actual_shape: tuple, expected_shape: tuple) -> bool:
+    """Compare the shape of two ndarrays.
+
+    Args:
+        actual_shape (tuple): the actual shape.
+        expected_shape (tuple): the expected shape.
+
+    Returns:
+        bool: returns true if the actual shape is the expected shape.
+    """
+    return all(a == e or e is None
+               for a, e in zip(actual_shape, expected_shape))
+
+
+def validate_spec(specs: dict, data: dict) -> bool:
+    """Validate whether the input data conform to the specs.
+
+    Args:
+        specs (dict): rules that should be followed.
+        data (dict): data to be evaluated.
+
+    Returns:
+        bool: returns true if the data follows the specs.
+    """
+    missing_keys = set(specs.keys()) - set(data.keys())
+    if missing_keys:
+        return False
+
+    for key, expected_shape in specs.items():
+        item = data[key]
+        if not validate_shape(item.shape, expected_shape):
+            return False
+    return True
+
+
+class SMPLDataConverter:
+    """A class that converts the input data into the smpl data."""
+    SMPL_DATA_SPECS = {
+        'betas': (1, 10),
+        'fullpose': (None, 24, 3),
+        'gender': (),
+        'mask': (None, ),
+        'transl': (None, 3)
+    }
+
+    SMPLX_DATA_SPECS = {
+        'betas': (1, 10),
+        'expression': (1, 10),
+        'fullpose': (None, 55, 3),
+        'gender': (),
+        'mask': (None, ),
+        'transl': (None, 3)
+    }
+
+    AMASS_SMPL_SPECS = {
+        'betas': (16, ),
+        'gender': (),
+        'poses': (None, 156),
+        'trans': (None, 3)
+    }
+
+    AMASS_SMPLX_SPECS = {
+        'betas': (16, ),
+        'gender': (),
+        'poses': (None, 165),
+        'trans': (None, 3)
+    }
+
+    HUMANDATA_SMPL_SPECS = {'meta': (), 'smpl': ()}
+
+    HUMANDATA_SMPLX_SPECS = {'meta': (), 'smplx': ()}
+
+    def __init__(self,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """
+        Args:
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be
+                selected. Defaults to None.
+        """
+        self.logger = logger
+
+    def get_data_type(self, filepath: str) -> str:
```

Comment:
`get_data_type`, `from_humandata` and `from_amass` looks more like static methods. Try static class and add this
```
def __new__(cls):
        raise NotImplementedError("SMPLDataConverter cannot be instantiated")
```